sys/hashes/sha3.c

```diff
@@ -0,0 +1,463 @@
+ */
+
+/*
+   ================================================================
+   The purpose of this source file is to demonstrate a readable and compact
+   implementation of all the Keccak instances approved in the FIPS 202 standard,
+   including the hash functions and the extendable-output functions (XOFs).
+
+   We focused on clarity and on source-code compactness,
+   rather than on the performance.
+
+   The advantages of this implementation are:
+ + The source code is compact, after removing the comments, that is. :-)
+ + There are no tables with arbitrary constants.
+ + For clarity, the comments link the operations to the specifications using
+        the same notation as much as possible.
+ + There is no restriction in cryptographic features. In particular,
+        the SHAKE128 and SHAKE256 XOFs can produce any output length.
+ + The code does not use much RAM, as all operations are done in place.
+
+   The drawbacks of this implementation are:
+    - There is no message queue. The whole message must be ready in a buffer.
+    - It is not optimized for peformance.
+
+   The implementation is even simpler on a little endian platform. Just define the
+   LITTLE_ENDIAN symbol in that case.
+
+   For a more complete set of implementations, please refer to
+   the Keccak Code Package at https://github.com/gvanas/KeccakCodePackage
+
+   For more information, please refer to:
+ * [Keccak Reference] http://keccak.noekeon.org/Keccak-reference-3.0.pdf
+ * [Keccak Specifications Summary] http://keccak.noekeon.org/specs_summary.html
+
+   This file uses UTF-8 encoding, as some comments use Greek letters.
+   ================================================================
+ */
+
+#include <hashes/sha3.h>
+
+/**
+ * Function to compute the Keccak[r, c] sponge function over a given input.
+ * @param  rate            The value of the rate r.
+ * @param  capacity        The value of the capacity c.
+ * @param  input           Pointer to the input message.
+ * @param  inputByteLen    The number of input bytes provided in the input message.
+ * @param  delimitedSuffix Bits that will be automatically appended to the end
+ *                         of the input message, as in domain separation.
+ *                         This is a byte containing from 0 to 7 bits
+ *                         These <i>n</i> bits must be in the least significant bit positions
+ *                         and must be delimited with a bit 1 at position <i>n</i>
+ *                         (counting from 0=LSB to 7=MSB) and followed by bits 0
+ *                         from position <i>n</i>+1 to position 7.
+ *                         Some examples:
+ *                             - If no bits are to be appended, then @a delimitedSuffix must be 0x01.
+ *                             - If the 2-bit sequence 0,1 is to be appended (as for SHA3-*), @a delimitedSuffix must be 0x06.
```

Comment:
Line length (see [coding conventions](https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions))

Reply:
Is there a reason why the line length is not handled by the uncrustify configuration?